File: src/main/java/darkorg/betterdurability/event/ForgeClientEventsHandler.java
package darkorg.betterdurability.event;

import darkorg.betterdurability.BetterDurability;
import darkorg.betterdurability.util.VanillaDamageableType;
import net.minecraft.ChatFormatting;
import net.minecraft.network.chat.Component;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.event.entity.player.ItemTooltipEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;

import java.util.List;

@Mod.EventBusSubscriber(modid = BetterDurability.MOD_ID, value = Dist.CLIENT)
public class ForgeClientEventsHandler {
    @SubscribeEvent
    public static void onItemTooltip(ItemTooltipEvent event) {
        List<Component> tooltip = event.getToolTip();
        if (VanillaDamageableType.isItemKnownBroken(event.getItemStack()) || VanillaDamageableType.isItemKnownBrokenAnother(event.getItemStack()) ) {
            tooltip.add(0, Component.translatable("tooltip.betterdurability.broken").withStyle(ChatFormatting.RED));
        }
    }
}

File: forge/shared/src/main/kotlin/darkorg/betterdurability/event/ForgeClientEventsHandler.kt
package darkorg.betterdurability.event

import darkorg.betterdurability.BetterDurability
import darkorg.betterdurability.polyfill.minecraft.text.TextFormatting
import darkorg.betterdurability.polyfill.minecraft.text.makeTranslatableComponent
import darkorg.betterdurability.util.VanillaDamageableType
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.event.entity.player.ItemTooltipEvent
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod.EventBusSubscriber

@EventBusSubscriber(modid = BetterDurability.MOD_ID, value = [Dist.CLIENT])
object ForgeClientEventsHandler {
    @SubscribeEvent
    fun onItemTooltip(event: ItemTooltipEvent) {
        val tooltip = event.toolTip
        if (VanillaDamageableType.isItemKnownBroken(event.itemStack)) {
            tooltip.add(0, makeTranslatableComponent("tooltip.betterdurability.broken").withStyle(TextFormatting.RED))
        }
    }
}
